featuretools/tests/primitive_tests/test_feature_serialization.py

```diff
@@ -94,3 +107,132 @@ def serialize_name_unchanged(original):
+    ft.save_features(features_original, TEST_S3_URL, profile_name=False)
+
+    obj = list(s3_bucket.objects.all())[0].key
+    s3_client.ObjectAcl(BUCKET_NAME, obj).put(ACL='public-read-write')
+
+    features_deserialized = ft.load_features(TEST_S3_URL, profile_name=False)
+
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+@pytest.fixture
+def setup_test_profile(monkeypatch):
+    test_path = os.path.join(CACHE, 'test_credentials')
+    test_path_config = os.path.join(CACHE, 'test_config')
+    monkeypatch.setenv("AWS_SHARED_CREDENTIALS_FILE", test_path)
+    monkeypatch.setenv("AWS_CONFIG_FILE", test_path_config)
+    monkeypatch.delenv("AWS_ACCESS_KEY_ID", raising=False)
+    monkeypatch.delenv("AWS_SECRET_ACCESS_KEY", raising=False)
+    monkeypatch.setenv("AWS_PROFILE", "test")
+
+    try:
+        os.remove(test_path)
+        os.remove(test_path_config)
+    except EnvironmentError:
+        pass
+
+    f = open(test_path, "w+")
+    f.write("[test]\n")
+    f.write("aws_access_key_id=AKIAIOSFODNN7EXAMPLE\n")
+    f.write("aws_secret_access_key=wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY\n")
+    f.close()
+    f = open(test_path_config, "w+")
+    f.write("[profile test]\n")
+    f.write("region=us-east-2\n")
+    f.write("output=text\n")
+    f.close()
+
+    yield
+    os.remove(test_path)
+    os.remove(test_path_config)
+
+
+def test_s3_test_profile(es, s3_client, s3_bucket, setup_test_profile):
+    features_original = ft.dfs(target_entity='sessions', entityset=es, features_only=True)
+
+    ft.save_features(features_original, TEST_S3_URL, profile_name='test')
+
+    obj = list(s3_bucket.objects.all())[0].key
+    s3_client.ObjectAcl(BUCKET_NAME, obj).put(ACL='public-read-write')
+
+    features_deserialized = ft.load_features(TEST_S3_URL, profile_name='test')
+
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_deserialize_features_default_s3(es):
+    # TODO: Feature ordering is different in py3.5 vs 3.6+
+    features_original = sorted(ft.dfs(target_entity='sessions', entityset=es, features_only=True), key=lambda x: x.unique_name())
+    features_deserialized = sorted(ft.load_features(S3_URL), key=lambda x: x.unique_name())
+    for feat_1, feat_2 in zip(features_original, features_deserialized):
+        assert feat_1.unique_name() == feat_2.unique_name()
+        assert feat_1.entityset == feat_2.entityset
+
+
+def test_features_anon_s3(es):
+    # TODO: Feature ordering is different in py3.5 vs 3.6+
+    features_original = sorted(ft.dfs(target_entity='sessions', entityset=es, features_only=True), key=lambda x: x.unique_name())
+    features_deserialized = sorted(ft.load_features(S3_URL, profile_name=False), key=lambda x: x.unique_name())
```

Comment:
the only thing that changes between test_deserialize_features_default_s3,  test_features_anon_s3, and test_deserialize_features_url is the url path and the profile.  We merge these into a single test that parametrizes url and profile as variables